### ISSUE TITLE
fitBounds() is whacky until resize() is called

### ISSUE THREAD

**Issue by stevage** [mapbox/mapbox-gl-js#11004]
**mapbox-gl-js version**: 2.3.1

[I submit this bug report for your benefit, not mine. It is not a complete repro.]

### Steps to Trigger Behavior

https://user-images.githubusercontent.com/678344/133007695-e236934e-abfb-4bcf-a1a6-55af2748d501.mov

 1. Call fitBounds() for a given bbox.
 2. The map pans to the wrong area - half the bounds is off screen.
 3. Call resize() and it's fine.

Note that the map is otherwise correctly sized within its container - all four corners are correctly located.

May be related to #8920, not sure.

**Comment by rreusser** [comment]
So that I can debug most effectively, what are the bounds of the map? A form covers half the page. Is the map under the form with padding to achieve the desired viewable area, or is the map container scaled to just the area in which the map is visible?

**Comment by stevage** [comment]
>Is the map under the form with padding to achieve the desired viewable area,

No, you can tell this is not the case because you can see the Mapbox logo in the bottom left.

I have deployed a dev version of the site here: http://mapswell.site44.com/11004/#dev

You can access the `map` object in the console.
